libraries/core/build.gradle.kts

```diff
@@ -30,6 +30,7 @@ java {
 dependencies {
     implementation(libs.coroutines.core)
     implementation(platform(libs.network.okhttp.bom))
+    implementation(libs.kotlin.reflect)
```

Comment:
Why do you need this lib? I'm not sure you're using it and I'm not sure we want to have it in the project (it will increase the size of the APK)

Reply:
> @jmartinesp was this merge intentional? It includes changes that do not seem related to this PR:
> 
> <img alt="image" width="818" src="https://user-images.githubusercontent.com/11990514/226560595-8ab023c1-6d8e-4c2c-afa2-226300d08364.png">

I added it so @callumu  could share the logs containing info about the `StateMachine` for the issues he mentioned,  but I should have reverted that yesterday. I'll do that now.